mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxLifecycleAwareObserver.kt

```diff
@@ -1,35 +1,40 @@
 package com.airbnb.mvrx
 
-import android.arch.lifecycle.*
+import android.arch.lifecycle.Lifecycle
 import android.arch.lifecycle.Lifecycle.Event
 import android.arch.lifecycle.Lifecycle.State
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.LifecycleOwner
+import android.arch.lifecycle.OnLifecycleEvent
 import io.reactivex.Observer
 import io.reactivex.disposables.Disposable
-import io.reactivex.functions.*
+import io.reactivex.functions.Action
+import io.reactivex.functions.Consumer
 import io.reactivex.internal.disposables.DisposableHelper
 import io.reactivex.internal.functions.Functions
 import io.reactivex.internal.observers.LambdaObserver
-import java.util.concurrent.atomic.*
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
 
 /**
  * An wrapper around an [Observer] associated with a [LifecycleOwner]. It has an [activeState], and when in a lifecycle state greater
  * than the [activeState] (as defined by [Lifecycle.State.isAtLeast()]) it will deliver values to the [sourceObserver] or [onNext] lambda.
  * When in a lower lifecycle state, the most recent update will be saved, and delivered when active again.
  */
 internal class MvRxLifecycleAwareObserver<T>(
-        owner: LifecycleOwner,
-        private val activeState: Lifecycle.State = DEFAULT_ACTIVE_STATE,
-        private val alwaysDeliverLastValueWhenUnlocked: Boolean = false,
-        private val sourceObserver: Observer<T>) : AtomicReference<Disposable>(), LifecycleObserver, Observer<T>, Disposable {
+    owner: LifecycleOwner,
```

Comment:
can we make the `owner` a `private var` and optional?

Reply:
Oh thanks for pointing this out. We can make the constructor take non-null types, but make the main constructor take private, nullable types.